Code/Framework/AzCore/AzCore/Debug/TraceReflection.cpp

```diff
@@ -48,63 +94,197 @@ namespace AZ
             bool OnPrintf(const char* window, const char* message) override;
             bool OnOutput(const char* window, const char* message) override;
 
+            // AZ::TickBus::Handler overrides ...
+            void OnTick(float deltaTime, AZ::ScriptTimePoint time) override;
+            int GetTickOrder() override;
+
         private:
-            template<class R, class... Args>
-            R CallResultReturn(const R& defaultReturnValue, int index, Args&&... args) const
+            void QueueMessageCall(AZStd::function<void()> messageCall);
+            void FlushMessageCalls();
+
+            AZStd::list<AZStd::function<void()>> m_messageCalls;
+            AZStd::mutex m_lock;
```

Comment:
I'd recommend a better name for this lock.
Something like m_messageCallListLock; ?

Reply:
Agreed, I will change it in a subsequent PR.

Reply:
Done